LXStudio-IDE/src/main/java/titanicsend/pattern/tom/Fire.java

```diff
@@ -0,0 +1,95 @@
+package titanicsend.pattern.tom;
+
+import heronarts.lx.LX;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.parameter.BooleanParameter;
+import heronarts.lx.parameter.BoundedParameter;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+import titanicsend.pattern.PeriodicPattern;
+
+import static processing.core.PApplet.lerpColor;
+
+public class Fire extends PeriodicPattern {
+    private float ROW_HEIGHT = 100000;
+    private float COLUMN_WIDTH = 100000;
+    private int NUM_COLUMNS;
+    private int NUM_ROWS;
+    private int[][] buffer;
+    private int[] gradient;
+
+    protected final CompoundParameter fuel = (CompoundParameter)
```

Comment:
Even though this is really a boolean parameter, I found the only way I could modulate if with LFOs, etc. in the UI was to set it up as a `CompoundParameter`. As you can see in the GIF, this allows us to do fancy stuff like trigger the fire every X beats.